cli/src/cmd/bind.rs

```diff
@@ -0,0 +1,301 @@
+use crate::cmd::Cmd;
+
+use clap::{Parser, ValueHint};
+use ethers::contract::Abigen;
+use eyre::{bail, eyre, WrapErr};
+use foundry_config::{
+    figment::{
+        self,
+        error::Kind::InvalidType,
+        value::{Dict, Map, Value},
+        Figment, Metadata, Profile, Provider,
+    },
+    find_project_root_path, Config,
+};
+use serde::Serialize;
+use std::{
+    io::Write,
+    path::{Path, PathBuf},
+};
+
+// Loads project's figment and merges the bind cli arguments into it
+impl<'a> From<&'a BindArgs> for Figment {
+    fn from(args: &'a BindArgs) -> Self {
+        let root = match args.root.clone() {
+            Some(root) => root,
+            None => find_project_root_path().unwrap(),
+        };
+
+        Config::figment_with_root(root).merge(args)
+    }
+}
+
+impl<'a> From<&'a BindArgs> for Config {
+    fn from(args: &'a BindArgs) -> Self {
+        let figment: Figment = args.into();
+        Config::from_provider(figment).sanitized()
+    }
+}
```

Comment:
this would be equivalent
```suggestion
// Loads project's figment and merges the bind cli arguments into it
foundry_config::impl_figment_convert!(BindArgs);
```

Reply:
tried that first and got a compilation error